extensions/spring-security/deployment/src/main/java/io/quarkus/spring/security/deployment/SpringSecurityProcessor.java

```diff
@@ -0,0 +1,74 @@
+package io.quarkus.spring.security.deployment;
+
+import java.lang.reflect.Modifier;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.AnnotationTarget;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.arc.deployment.InterceptorBindingRegistrarBuildItem;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ApplicationIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.security.deployment.AdditionalSecurityCheckBuildItem;
+import io.quarkus.security.deployment.SecurityCheckInstantiationUtil;
+import io.quarkus.spring.security.runtime.interceptor.SpringSecuredInterceptor;
+
+class SpringSecurityProcessor {
+
+    private static final String FEATURE = "spring-security";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    void registerSecurityInterceptors(BuildProducer<InterceptorBindingRegistrarBuildItem> registrars,
+            BuildProducer<AdditionalBeanBuildItem> beans) {
+        registrars.produce(new InterceptorBindingRegistrarBuildItem(new SpringSecurityAnnotationsRegistrar()));
+        beans.produce(new AdditionalBeanBuildItem(SpringSecuredInterceptor.class));
+    }
+
+    @BuildStep
+    void addSpringSecuredSecurityCheck(ApplicationIndexBuildItem index,
+            BuildProducer<AdditionalSecurityCheckBuildItem> additionalSecurityCheckBuildItems) {
+
+        Set<MethodInfo> methodsWithSecurityAnnotation = new HashSet<>();
+        for (AnnotationInstance instance : index.getIndex().getAnnotations(DotNames.SPRING_SECURED)) {
```

Comment:
it looks like we're assuming that noone mixes `@Secured` with the spec annotations. 
I think it would be good to validate if it's true before making this assumption.
Or not add the class-level annotation to methods that have these annotations.

Reply:
I would say that at this point we should probably not complicate things with mix and match. We can maybe just log a warning if a spec annotation is used?

Reply:
+1 for the warning.

Reply:
not sure if a warning is enough...
The current logic would add a duplicate security annotation to a method in such case, wouldn't it?

Reply:
Well if people go an mix everything they can of course end up with weird situations. I think that a clear warning that Spring and spec annotations should not be mixed (on classes or methods) should be enough.
I don't think we want to start handling cases where Spring annotations are mixed with other annotations, unless Spring itself supports this. It could unecessarily complicate things I think.

Reply:
agree on not supporting a mix.
Still not sure if warning is enough.
Do we even know what happens if a user has `@Secured("user") on the class and `@DenyAll` on its method?
I think it will fail in the runtime having two checks for a single method, but I'm not sure (this is how it worked before latest changes).
IMO a compile time error would be much better in such case.

Reply:
A build time error sounds good indeed. Let's do that, OK @aureamunoz ?

Reply:
Ok for the build time error, let's do it.

Reply:
This was fixed